internal/resource/factory.go

```diff
@@ -64,11 +64,18 @@ func getManager() Manager {
 	if hasNVML {
 		klog.Info("Using NVML manager")
 		return NewNVMLManager()
-	} else if isTegra {
+	}
+
+	if isTegra {
 		klog.Info("Using CUDA manager")
 		return NewCudaManager()
 	}
 
+	if *config.Flags.GFD.VMPassthroughWorkload {
```

Comment:
I agree with Evan's comment: https://github.com/NVIDIA/k8s-device-plugin/pull/676/files#r1575766699

We should add a new `vfio` mode to the `go-nvlib/info` package, which represents a system where all GPUs are bound to the `vfio-pci` driver. We can construct a `VfioManager` if we detect a `vfio` system, so no need to introduce a new `VMPassthroughWorkload` flag.

Reply:
To clarify. The `mode` flag I suggested in the linked comment is local. We could (and probably should) pull some of the functionality into the `go-nvlib/info` package.

The important functionality from the perspective of GFD and the device plugin is that a user can explicitly override the mode if our autodetection fails. This already happens on newer Tegra-based systems, for example.

Reply:
Added this